src/query/ast/src/ast/expr.rs

```diff
@@ -212,7 +212,10 @@ pub enum Expr {
     /// The `Array` expr
     Array { span: Span, exprs: Vec<Expr> },
     /// The `Map` expr
-    Map { span: Span, kvs: Vec<(Expr, Expr)> },
+    Map {
```

Comment:
Key in `Expr` indeed causes ambiguity, but why can't values be Expr?

Reply:
@ariesdevil

Reply:
Wouldn't sql like `select {num: number + 1} from numbers(100)` be useful?

Reply:
Seems no user using it that way. BTW: duckdb supports using expression for both key/value, but they don't support using `:` to access.

Reply:
I mean, is there a simple way to construct bunch of maps based on query results. A more practical example would be `INSERT INTO orders_json_data (data) SELECT {order_id: order_id, order_timestamp: timestamp}` from old_order join ....`.

Reply:
I think it's better to have, we can open an new issue here(may not in high-priority), also cc @b41sh